dir.targets

```diff
@@ -88,11 +88,13 @@
     <Copy Condition="Exists('$(NuGetConfigFile)')" SourceFiles="$(NuGetConfigFile)" DestinationFolder="$(PackagesDir)" SkipUnchangedFiles="true" />
     <Copy Condition="Exists('$(NuGetConfigFile)')" SourceFiles="$(NuGetConfigFile)" DestinationFolder="$(IntermediateOutputRootPath)" SkipUnchangedFiles="true" />
 
-    <!-- Add DNU execute rights -->
+    <!-- Add DNU and Roslyn tool execute rights -->
     <Exec Condition="'$(OsEnvironment)'=='Unix'"
           Command="chmod a+x &quot;$(DnxPackageDir)/bin/dnu&quot;" />
     <Exec Condition="'$(OsEnvironment)'=='Unix'"
           Command="chmod a+x &quot;$(DnxPackageDir)/bin/dnx&quot;" />
+    <Exec Condition="'$(OsEnvironment)'=='Unix'"
+          Command="find '$(RoslynPackageDir)tools' -name &quot;*.exe&quot; -exec chmod &quot;+x&quot; '{}' ';'" /> 
```

Comment:
There are a number of ways tasks can execute external program.  I use mono to run programs that end with .exe, but I only changed Exec task and ToolTask.Execute method. Any task that overrides that method will need to be modified. The alternative is to give execute permission to the .exe file.

From: Jared Parsons [mailto:notifications@github.com]
Sent: Friday, May 01, 2015 8:23 AM
To: dotnet/corefx
Cc: Val Menn
Subject: Re: [corefx] Workarounds for building with Roslyn/MSBuild on Unix. (#1601)


In dir.targets<https://github.com/dotnet/corefx/pull/1601#discussion_r29507778>:

>      <Exec Condition="'$(OsEnvironment)'=='Unix'"

>            Command="chmod a+x &quot;$(DnxPackageDir)/bin/dnu&quot;" />

>      <Exec Condition="'$(OsEnvironment)'=='Unix'"

>            Command="chmod a+x &quot;$(DnxPackageDir)/bin/dnx&quot;" />

> +    <Exec Condition="'$(OsEnvironment)'=='Unix'"

> +          Command="find '$(RoslynPackageDir)tools' -name &quot;*.exe&quot; -exec chmod &quot;+x&quot; '{}' ';'" />

Curious. Why was this needed? In general the compilers are not executed directly but instead by being prefixed with mono csc.exe.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/1601/files#r29507778>.